File: Data Structures & Algorithms/binary-tree-postorder-traversal/submission-0.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func postorderTraversal(root *TreeNode) []int {
    if root == nil {
		return []int{}
	}

	res := make([]int, 0)
	left := postorderTraversal(root.Left)
	res = append(res, left...)
	right := postorderTraversal(root.Right)
	res = append(res, right...)
	res = append(res, root.Val)
	return res
}

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-0.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @param {number} k
     * @return {number}
     */
    kthSmallest(root, k) {
        let res = []
        this.dfs(root, res)
       
        return res.sort()[k-1]
    }

    dfs(root,res){
        if(!root) {return}
        res.push(root.val)

        this.dfs(root.left,res)
        this.dfs(root.right, res)
    }
}
